parley/src/tests/utils/env.rs

```diff
@@ -0,0 +1,227 @@
+use std::sync::Once;
+use tiny_skia::Pixmap;
+
+static INIT: Once = Once::new();
+
+/// Initialize tests
+/// Removes all images from directory with the current test images
+pub(crate) fn initialize_tests() {
+    INIT.call_once(|| {
+        let entries = std::fs::read_dir(current_imgs_dir()).unwrap();
+        for entry in entries.flatten() {
+            let path = entry.path();
+            if path
+                .extension()
+                .and_then(|ext| ext.to_str())
+                .map(|ext| ext == "png")
+                .unwrap_or(false)
+            {
+                std::fs::remove_file(&path).unwrap();
+            }
+        }
+    });
+}
+
+fn is_accept_mode() -> bool {
+    std::env::var("PARLEY_TEST")
+        .map(|x| x.to_ascii_lowercase() == "accept")
+        .unwrap_or(false)
+}
+
+pub(crate) fn load_fonts_dir(collection: &mut Collection, path: &Path) -> std::io::Result<()> {
+    let paths = std::fs::read_dir(path)?;
+    for entry in paths {
+        let entry = entry?;
+        if !entry.metadata()?.is_file() {
+            continue;
+        }
+        let path = entry.path();
+        if path
+            .extension()
+            .and_then(|ext| ext.to_str())
+            .map(|ext| !["ttf", "otf", "ttc", "otc"].contains(&ext))
+            .unwrap_or(true)
+        {
+            continue;
+        }
+        let font_data = std::fs::read(&path)?;
+        collection.register_fonts(font_data);
+    }
+    Ok(())
+}
+
+impl TestEnv {
+    pub(crate) fn new(test_name: &str) -> Self {
+        initialize_tests();
+        let mut collection = Collection::new(CollectionOptions {
+            shared: false,
+            system_fonts: false,
+        });
+        load_fonts_dir(&mut collection, &font_dir()).unwrap();
+        collection
+            .family_id(DEFAULT_FONT_NAME)
+            .unwrap_or_else(|| panic!("{} font not found", DEFAULT_FONT_NAME));
+        TestEnv {
+            test_name: test_name.to_string(),
+            check_counter: 0,
+            font_cx: FontContext {
+                collection,
+                source_cache: Default::default(),
+            },
+            tolerance: 0.0,
+            layout_cx: LayoutContext::new(),
+            foreground_color: Color::rgb8(0, 0, 0),
+            background_color: Color::rgb8(255, 255, 255),
+            errors: Vec::new(),
+        }
+    }
+
+    pub(crate) fn builder<'a>(&'a mut self, text: &'a str) -> RangedBuilder<'a, Color> {
+        let mut builder = self.layout_cx.ranged_builder(&mut self.font_cx, text, 1.0);
+        builder.push_default(StyleProperty::Brush(self.foreground_color));
+        builder.push_default(StyleProperty::FontStack(FontStack::Single(
+            FontFamily::Named(DEFAULT_FONT_NAME.into()),
+        )));
+        builder
+    }
+
+    fn image_name(&mut self) -> String {
+        let name = format!("{}-{}.png", self.test_name, self.check_counter);
+        self.check_counter += 1;
+        name
```

Comment:
My preference would be for the name of a test to be passed explicitly.
I say this both so that adding a new test early doesn't move all of the existing tests out of sync, and so that the image names are self descriptive.
If we ended up duplicating the cool side-by-side view you showed us on Zulip, this would also be really nice there. (But that's definitely not a job for this PR).

However, this is a tradeoff, because naming tests is hard. Leaving it as is would be fine for this PR.

Reply:
The name of the image is automatically generated from the name of the test, so it it easy to find a corresponding test that generates the image. The counter is used for solving the situation that a test generates more images. 

But maybe I am misunderstanding this comment.

Reply:
So specifically, my preference would be for the format to be `{test_name}-{snapshot_name}`; i.e. each snapshot gets a descriptive name (each test in turn could then use their own indexing if they wanted to).

Reply:
I like the idea and I originally planed it put it in this PR, but I have stripped it to make it more straightforward.
I added `check_snapshot_with_name`, it is a simple change.